@here/harp-datasource-protocol/test/ExprEvaluatorTest.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// tslint:disable:only-arrow-functions
+//    Mocha discourages using arrow functions, see https://mochajs.org/#arrow-functions
+
+import { assert } from "chai";
+import { Expr, MapEnv, ValueMap } from "../lib/Expr";
+import { ExprEvaluator } from "../lib/ExprEvaluator";
+
+const evaluator = new ExprEvaluator();
+
+const defaultEnv = {
+    on: true,
+    off: false,
+    someText: "some text",
+    emptyText: ""
+};
+
+function evaluate(expr: unknown, env: ValueMap = defaultEnv) {
+    return evaluator.evaluate(Expr.fromJSON(expr), new MapEnv(env));
+}
+
+describe("Operator 'all'", function() {
+    it("evaluate", function() {
+        assert.isTrue(
+            Boolean(
+                evaluate([
+                    "all",
+                    true,
+                    1,
+                    "string literal",
+                    ["has", "on"],
+                    ["get", "on"],
+                    ["has", "off"],
+                    ["!", ["get", "off"]]
+                ])
+            )
+        );
+
+        assert.isFalse(Boolean(evaluate(["all", true, 1, "string literal", ["get", "flag"]])));
+    });
+});
+
+describe("Operator 'any'", function() {
+    it("evaluate", function() {
+        assert.isTrue(Boolean(evaluate(["any", 1, true, "string", ["get", "on"]])));
+        assert.isFalse(Boolean(evaluate(["any", 0, false, "", ["get", "off"]])));
+    });
+});
+
+describe("Operator 'get'", function() {
+    Object.getOwnPropertyNames(defaultEnv).forEach((property: any) => {
+        const propertyName = property as keyof typeof defaultEnv;
+        it(`get property '${propertyName}'`, function() {
+            assert.strictEqual(evaluate(["get", propertyName]), defaultEnv[propertyName]);
+        });
+    });
+
+    it("Ensure builtin symbols are not accessible", function() {
+        assert.isUndefined(evaluate(["get", "has"]));
```

Comment:
From test description i would assume it checks `hasOwnProperty` and `constructor` are not accessible from expr context as this may constitute some kind of backdoor.

Reply:
`has` is using the `lookup` methods from the environment so it depends on how the environment is defined (MapEnv, other possible impelmentation of Env).
The point of this test is to make sure that when the parser (`Expr.fromJSON`) create nodes for `get` and `has` it doesn't expand the property name.